src/server/auth/server/http_client.go

```diff
@@ -0,0 +1,20 @@
+package server
+
+import (
+	"net/http"
+)
+
+type RewriteRoundTripper struct {
+	RewriteHost string
+}
+
+func RewriteClient(rewriteHost string) *http.Client {
```

Comment:
Returns an `http.Client` we can inject into the OIDC and OAuth libraries which silently rewrites requests where the host is `Issuer` to `IssuerOverride`.